mmdet/datasets/coco_with_text.py

```diff
@@ -271,14 +282,21 @@ def evaluate(self,
             cocoEval = COCOeval(cocoGt, cocoDt, iou_type)
             cocoEval.params.catIds = self.cat_ids
             cocoEval.params.imgIds = self.img_ids
-            lexicon_path = metric_params['lexicon']
-            lexicon = self._read_lexicon(lexicon_path)
-
-            lexicon_pairs_path = metric_params['lexicon_mapping']
-            lexicon_mapping = self._read_lexicon_mapping(lexicon_pairs_path)
 
             predictions = []
-            for res in tqdm(results):
+            for img_i, res in tqdm(enumerate(results)):
+                img_id = os.path.basename(cocoGt.imgs[img_i]['filename']).split('.')[0].split('_')[-1]
+
+                lexicon_path = metric_params['lexicon']
+                if lexicon_path and '{}' in lexicon_path:
+                    lexicon_path = lexicon_path.format(img_id)
+                lexicon = self._read_lexicon(lexicon_path)
```

Comment:
So, will lexicon be loaded from file on each iteration of the loop even what the lexicon global for all images? Or it is handled effectively somewhere?

Reply:
no, it was not.

Cached.